cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/MetricsProvider.java

```diff
@@ -30,5 +30,5 @@ public interface MetricsProvider {
    *
    * @param runId The {@link ProgramRunId} which references this execution.
    */
-  ExecutionMetrics[] retrieveMetrics(ProgramRunId runId);
+  ExecutionMetrics[] retrieveMetrics(ProgramRunId runId) throws Exception;
```

Comment:
Generally throwing `Exception` is an antipattern. Please throw some specific exception